unload_databricks_data_to_s3_partition.py

```diff
@@ -0,0 +1,158 @@
+    parser.add_argument("--secret_key_name_for_sql", required=True,
+                        help="databricks secret key name of transformation sql")
+    parser.add_argument("--s3_endpoint", required=True, help="s3 endpoint")
+    parser.add_argument("--s3_path", required=True, help="s3 path where data will be written into")
+    parser.add_argument("--max_records_per_file", help="max records per output file", nargs='?', type=int,
+                        default=MAX_RECORDS_PER_OUTPUT_FILE, const=MAX_RECORDS_PER_OUTPUT_FILE)
+    parser.add_argument("--ingestion_in_mutability_mode",
+                        help="""if provided, will not apply filter to exclude change data for some mutation actions.
+                        Otherwise, will include append-only (i.e. insert) for event data and upsert-only (i.e. insert
+                        and update_postimage) for user/group properties. The filter is enabled by default.""",
+                        action='store_true', default=False)
+
+    args, unknown = parser.parse_known_args()
+
+    spark = SparkSession.builder.getOrCreate()
+    # setup s3 credentials for data export
+    aws_access_key = dbutils.secrets.get(scope=args.secret_scope, key=args.secret_key_name_for_aws_access_key)
+    aws_secret_key = dbutils.secrets.get(scope=args.secret_scope, key=args.secret_key_name_for_aws_secret_key)
+    aws_session_token = dbutils.secrets.get(scope=args.secret_scope, key=args.secret_key_name_for_aws_session_token)
+    spark.conf.set("fs.s3a.aws.credentials.provider", "org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider")
+    spark.conf.set("fs.s3a.access.key", aws_access_key)
+    spark.conf.set("fs.s3a.secret.key", aws_secret_key)
+    spark.conf.set("fs.s3a.session.token", aws_session_token)
+    spark.conf.set("fs.s3a.endpoint", args.s3_endpoint)
+
+    sql: str = dbutils.secrets.get(scope=args.secret_scope, key=args.secret_key_name_for_sql)
+
+    # Build temp views
+    table_to_import_version_range_map: dict[str, list[int]] = parse_table_versions_map_arg(args.table_versions_map)
+    for table, import_version_range in table_to_import_version_range_map.items():
+        data: DataFrame = fetch_data(table, import_version_range[0], import_version_range[1])
+
+        if not args.ingestion_in_mutability_mode:
+            data = filter_data(data, args.data_type)
+
+        view_name: str = build_temp_view_name(table)
+        data.createOrReplaceTempView(view_name)
+        # replace table name in sql to get prepared for sql transformation
+        sql = sql.replace(table, view_name)
+
+    # run SQL to transform data
+    export_data: DataFrame = spark.sql(sql)
+
+    event_count = export_data.count()
+    partition_count = get_partition_count(event_count, args.max_records_per_file)
+
+    # export data
+    export_data.repartition(partition_count).write.mode("overwrite").option("maxRecordsPerFile", MAX_RECORDS_PER_OUTPUT_FILE).json(args.s3_path)
```

Comment:
`.repartition(partition_count)` do we need to do repartition if we set `maxRecordsPerFile`? concerned as @fzqgriffin mentioned repartitioning is expensive

Reply:
btw, could be the value of `max_records_per_file` argument instead of constant for `maxRecordsPerFile`

Reply:
no, it only matters if there are a ton of tiny files (which is probably unlikely)